mne/morph.py

```diff
@@ -0,0 +1,715 @@
+        """
+        import nibabel as nib
+        from dipy.align.reslice import reslice
+
+        shape = tuple([int(i) for i in self.morph_data['morph_shape']])
+        affine = self.morph_data['morph_affine']
+        zooms = self.morph_data['morph_zooms'][:3]
+
+        hdr = nib.nifti1.Nifti1Header()
+        hdr.set_xyzt_units('mm', 'msec')
+        hdr['pixdim'][4] = 1e3 * stc.tstep
+
+        new_zooms = None
+
+        if isinstance(mri_resolution, bool) and mri_resolution:
+            new_zooms = self.morph_data['mri_zooms']
+
+        elif isinstance(mri_resolution, tuple):
+            new_zooms = mri_resolution
+
+        img = np.zeros(shape).reshape(-1, stc.shape[1])
+        img[stc.vertices, :] = stc.data
+
+        img = img.reshape(shape + (-1,))
+
+        img = nib.Nifti1Image(img, affine, header=hdr)
+
+        if new_zooms is not None:
+            new_zooms = new_zooms[:3]
+            img, affine = reslice(img.get_data(),
+                                  img.affine,
+                                  zooms,
+                                  new_zooms)
+            img = nib.Nifti1Image(img, affine)
+            zooms = new_zooms
+        img.header.set_zooms(zooms + (1,))
+        if fname is not None:
+            nib.save(img, fname)
+        return img
+
+
+@verbose
+def read_source_morph(fname, verbose=None):
+    """Load the morph for source estimates from a file.
+
+    Parameters
+    ----------
+    fname : str
+        Full filename including path
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see
+        :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+        for more). Defaults to self.verbose.
+
+    Returns
+    -------
+    source_morph : instance of SourceMorph
+        The loaded morph.
+    """
+    try:
+        logger.info('loading morph...')
+        morph_data = read_hdf5(fname)
+    except IOError:
+        raise IOError('cannot read file: %s' % fname)
+
+    source_morph = SourceMorph(None)
+    source_morph.__dict__ = morph_data
+    logger.info('[done]')
+    return source_morph
+
+
+def _compute_morph_data(morph, src, verbose=None):
+    """Compute source estimate specific morph."""
+
+    morph_data = []
+
+    # VolSourceEstimate
+    if morph.kind == 'volume':
+
+        logger.info('volume source space inferred...')
+
+        import nibabel as nib
+
+        # load moving mri
+        mri_subpath = op.join('mri', 'brain.mgz')
+        mri_path_from = op.join(morph.subjects_dir, morph.subject_from,
+                                mri_subpath)
+
+        logger.info('loading %s as moving volume' % mri_path_from)
+        mri_from = nib.load(mri_path_from)
+
+        # load static mri
+        static_path = op.join(morph.subjects_dir, morph.subject_to)
+
+        if not op.isdir(static_path):
+            mri_path_to = static_path
+        else:
+            mri_path_to = op.join(static_path, mri_subpath)
+        try:
```

Comment:
avoid try-catch pattern when you can. Here, for instance you can check if the file exists using `op.isfile` ...